extensions/imports/__tests__/addImportCJSBasicConfig-test.js

```diff
@@ -0,0 +1,288 @@
+'use strict';
+
+jest.autoMockOff();
+
+const jscodeshift = require('jscodeshift');
+const imports = require('../index');
+
+const {statement} = jscodeshift.template;
+
+const DEFAULT_RECAST_CONFIG = {
+  quote: 'single',
+  trailingComma: true,
+  tabWidth: 2,
+};
+
+function test(inputCollection, output) {
+  expect(
+    (inputCollection.toSource(DEFAULT_RECAST_CONFIG) || '').trim()
+  ).toEqual(
+    (output || '').trim()
+  );
+}
+
+// Setup JSCodeShift
+const plugins = imports.createPlugins(imports.config.CJSBasicRequire);
+jscodeshift.registerMethods({
+  addBasicImport: plugins.addImport,
+});
+
+describe('addImportCJSBasicConfig', () => {
+  it('should add to block start', () => {
+    const jfile = jscodeshift(`
+      var A1 = require('A1');
+      var A3 = require('A3');
+
+      aaa;
+    `)
+    .addBasicImport(statement`
+      var A0 = require('A0');
+    `);
+
+    test(jfile, `
+      var A0 = require('A0');
+      var A1 = require('A1');
+      var A3 = require('A3');
+
+      aaa;
+    `);
+  });
+
+  it('should add to block middle', () => {
+    const jfile = jscodeshift(`
+      var A1 = require('A1');
+      var A3 = require('A3');
+
+      aaa;
+    `)
+    .addBasicImport(statement`
+      var A2 = require('A2');
+    `);
+
+    test(jfile, `
+      var A1 = require('A1');
+      var A2 = require('A2');
+      var A3 = require('A3');
+
+      aaa;
+    `);
+  });
+
+  it('should add to block end', () => {
+    const jfile = jscodeshift(`
+      var A1 = require('A1');
+      var A3 = require('A3');
+
+      aaa;
+    `)
+    .addBasicImport(statement`
+      var A4 = require('A4');
+    `);
+
+    test(jfile, `
+      var A1 = require('A1');
+      var A3 = require('A3');
+      var A4 = require('A4');
+
+      aaa;
+    `);
+  });
+
+  // TODO: Comments mess with the whitespace output, fix this.
```

Comment:
Comments are the worst, this is another reason why you shouldn't comment code ever /s